scripts/tools/silabs/FactoryDataProvider.py

```diff
@@ -0,0 +1,330 @@
+        if self._args.gen_spake2p_path:
+            self._args.spake2_verifier = self.generate_spake2p_verifier()
+
+    def add_SerialNo_To_CMD(self, cmdList):
+        """ Add the jtag serial command to the commander command
+
+            Args:
+            The commander command in list format
+        """
+        if self._args.jtag_serial:
+            cmdList.extend(["--serialno", self._args.jtagSerial])
+
+    def create_nvm3injected_image(self):
+        """ Use commander command lines create a binary flashable to the EFR32 
+            containing the factory commissioning data in NVM3 section
+        """
+        isDeviceConnected = True
+
+        # Retrieve the device current nvm3 data in a binary file
+        # It will be used as base to add the new credentials
+        inputImage = self.TEMP_FILE
+        cmd = ['commander', 'nvm3', 'read', '-o', inputImage, ]
+        self.add_SerialNo_To_CMD(cmd)
+        results = subprocess.run(cmd, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+
+        if results.returncode != 0:
+            # No nvm3 section found. Retrieve the device info
+            cmd = ['commander', 'device', 'info', ]
+            self.add_SerialNo_To_CMD(cmd)
+            try:
+                output = subprocess.check_output(cmd)
+                output = output.decode('utf-8').splitlines()
+                deviceInfo = dict(map(str.strip, lines.split(':')) for lines in output[0:len(output)-1])
+                # Only MG12 and MG24 are supported in matter currently
+                if "EFR32MG12" in deviceInfo["Part Number"]:
+                    inputImage = self.BASE_MG12_FILE
+                elif "EFR32MG24" in deviceInfo["Part Number"]:
+                    inputImage = self.BASE_MG24_FILE
+                else:
+                    raise Exception('Invalid MCU')
+            except:
+                isDeviceConnected = False
+                print("Device not connected")
+                # When no device is connected user needs to provide the mcu family for which those credentials are to be created
+                if self._args.mcu_family:
+                    if "EFR32MG12" == self._args.mcu_family:
+                        inputImage = self.BASE_MG12_FILE
+                    elif "EFR32MG24" == self._args.mcu_family:
+                        inputImage = self.BASE_MG24_FILE
+                else:
+                    print("Connect debug port or provide the mcu_family")
+                    return
+
+        # Convert interger to little endian hex format and strings to hex byte array format for nvm3 storage
+        Passcode = self._args.passcode.to_bytes(4, "little").hex()
+        Spake2pIterationCount = self._args.spake2_iteration.to_bytes(4, 'little').hex()
+        Discriminator = self._args.discriminator.to_bytes(2, 'little').hex()
+        SaltByteArray = bytes(self._args.spake2_salt, 'utf-8').hex()
+        VerifierByteArray = bytes(self._args.spake2_verifier, 'utf-8').hex()
```

Comment:
Convention is that only class names in python start with a capital letter.

Reply:
Ok will change